python/pdo/submitter/sawtooth/sawtooth_submitter.py

```diff
@@ -0,0 +1,430 @@
+        enclave_signature,
+        contract_id,
+        message_hash,
+        current_state_hash,
+        contract_code_hash,
+        **extra_params):
+
+        json_input = JsonPayloadBuilder.build_ccl_transaction_from_data(
+            self.pdo_signer.signing_key,
+            self.pdo_signer.verifying_key,
+            'initialize',
+            channel_keys.txn_public,
+            contract_enclave_id,
+            enclave_signature,
+            contract_id,
+            crypto.byte_array_to_base64(message_hash),
+            crypto.byte_array_to_base64(current_state_hash),
+            "",     # previous_state_hash,
+            "",     # encyrpted root block. No longer stored in Sawtooth
+            [],     # empty dependency_list
+            crypto.byte_array_to_base64(contract_code_hash))
+                            # contract code hash is necessary for the pdo signature
+
+        extra_params['key_str'] = channel_keys.txn_private
+
+        return self.submit_json(json_input, json_input['af'], **extra_params)
+
+# -----------------------------------------------------------------
+    def ccl_update(self,
+        channel_keys,
+        contract_enclave_id,
+        enclave_signature,
+        contract_id,
+        message_hash,
+        current_state_hash,
+        previous_state_hash,
+        dependency_list,
+        **extra_params):
+
+        json_input = JsonPayloadBuilder.build_ccl_transaction_from_data(
+            "",     #no creator private key, so no pdo signature included
+            "",     #no need for creator public key for update txns
+            'update',
+            channel_keys.txn_public,
+            contract_enclave_id,
+            enclave_signature,
+            contract_id,
+            crypto.byte_array_to_base64(message_hash),
+            crypto.byte_array_to_base64(current_state_hash),
+            crypto.byte_array_to_base64(previous_state_hash),
+            "",     #encyrpted root block. No longer stored in Sawtooth
+            dependency_list,
+            "")  #no contract hash because there is no pdo sign
+
+        extra_params['key_str'] = channel_keys.txn_private
+
+        return self.submit_json(json_input, json_input['af'], **extra_params)
+
+# -----------------------------------------------------------------
+    def get_enclave_info(self,
+        enclave_id):
+
+        return self.read_helper.get_enclave_dict(enclave_id)
+
+# -----------------------------------------------------------------
+    def get_contract_info(self,
+        contract_id):
+
+        return self.read_helper.get_contract_dict(contract_id)
+
+# -----------------------------------------------------------------
+    def get_current_state_hash(self,
+        contract_id):
+
+        ccl_info = self.read_helper.get_ccl_info_dict(contract_id)
+
+        state_info = dict()
+        state_info['state_hash'] = ccl_info['current_state']['state_hash']
+        state_info['is_active'] = ccl_info['is_active']
```

Comment:
Why do we need to return `is_active` here? The CCF equivalent only returns the hash.

Reply:
I would not want to change existing APIs. 'is_active' is part of the response in Sawtooth, we are simply renaming the output variable so as to use a single API for both ccf and sawtooth, ccf also returns is_active.